app/models/user.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :sent_friend_requests, -> { where('status = 0') },
+           class_name: 'Friendship', foreign_key: 'from_user_id'
+  has_many :sent_friend_requests_users, through: :sent_friend_requests,
+                                        source: :to_user
+
+  has_many :received_friend_requests, -> { where('status = 0') },
+           class_name: 'Friendship', foreign_key: 'to_user_id'
+  has_many :received_friend_requests_users, through: :received_friend_requests,
+                                            source: :from_user
+
+  has_many :requested_friendships, -> { where('status = 1') },
+           class_name: 'Friendship', foreign_key: 'from_user_id'
+  has_many :requested_friends, through: :requested_friendships,
+                               source: :to_user
+
+  has_many :accepted_friendships, -> { where('status = 1') },
+            class_name: 'Friendship', foreign_key: 'to_user_id'
+  has_many :accepted_friends, through: :accepted_friendships,
+                              source: :from_user
+
+  has_many :author_posts, class_name: 'Post', foreign_key: 'post_id'
+
+  has_many :comments
+  has_many :commented_posts, through: :comments,
+            source: :subject, source_type: 'Post'
+  has_many :commented_comments, through: :comments,
+            source: :subject, source_type: 'Comment'
+
+  has_many :likes
+  has_many :liked_posts, through: :likes, source: :subject, source_type: 'Post'
```

Comment:
We might be overdoing associations here. Probably for posts all the association we need to have is for those that were created/authored by the user:

```ruby
has_many :posts, foreign_key: user_id, dependent: :destroy
```

For `liked` posts etc we can simply have instance methods for them. I'm thinking about it though, why do we need to have such special associations/methods even for a `user` 's liked posts, etc? For the starting phases of our app building we better stick to the more simpler methods/associations first then we add features as we go along.